File: jdroid-android-core/src/main/java/com/jdroid/android/search/ExtendedPagedResult.java
package com.jdroid.android.search;

import com.jdroid.java.collections.Lists;
import com.jdroid.java.search.PagedResult;

import java.util.Collection;
import java.util.List;

/**
 * @param <T> The list item.
 */
public class ExtendedPagedResult<T> extends PagedResult<T> {

	private List<T> extraResults = Lists.newArrayList();

	/**
	 * Adds a result item to the extra list.
	 *
	 * @param result The result to add.
	 */
	public void addExtraResult(T result) {
		extraResults.add(result);
	}

	/**
	 * Adds result items to the extra list.
	 *
	 * @param results The results to add.
	 */
	public void addExtraResults(Collection<T> results) {
		extraResults.addAll(results);
	}

	/**
	 * @return the extraResults
	 */
	public List<T> getExtraResults() {
		return extraResults;
	}
}

File: jdroid-android-core/src/main/java/com/jdroid/android/search/ExtendedPagedResult.kt
package com.jdroid.android.search

import com.jdroid.java.collections.Lists
import com.jdroid.java.search.PagedResult

/**
 * @param <T> The list item.
</T> */
class ExtendedPagedResult<T> : PagedResult<T>() {

    private val extraResults = Lists.newArrayList<T>()

    /**
     * Adds a result item to the extra list.
     *
     * @param result The result to add.
     */
    fun addExtraResult(result: T) {
        extraResults.add(result)
    }

    /**
     * Adds result items to the extra list.
     *
     * @param results The results to add.
     */
    fun addExtraResults(results: Collection<T>) {
        extraResults.addAll(results)
    }

    /**
     * @return the extraResults
     */
    fun getExtraResults(): List<T> {
        return extraResults
    }
}
